embassy-nrf/src/uarte.rs

```diff
@@ -487,19 +568,23 @@ impl<'d, U: Instance, T: TimerInstance> Write for UarteWithIdle<'d, U, T> {
 }
 
 pub(crate) mod sealed {
+    use core::sync::atomic::AtomicU8;
+
     use embassy::waitqueue::AtomicWaker;
 
     use super::*;
 
     pub struct State {
         pub endrx_waker: AtomicWaker,
         pub endtx_waker: AtomicWaker,
+        pub tx_rx_refcount: AtomicU8,
     }
     impl State {
         pub const fn new() -> Self {
             Self {
                 endrx_waker: AtomicWaker::new(),
                 endtx_waker: AtomicWaker::new(),
+                tx_rx_refcount: AtomicU8::new(2),
```

Comment:
Make this `0`, so the states can go in `.bss` so it doesn't use extra flash.

Set it to `2` in `Uart::new()`. it's needed for correctness too: if you drop the uart then create it again refcount has to be reset to 2, otherwise the second drop wont work.

Reply:
Fixed in commit 0642eec01e23d9037e21186cfb0c169e56f9a161

Reply:
:rocket: !!!